algorithms/scripts/WIP-passiveaggressive-KT12.py

```diff
@@ -0,0 +1,310 @@
+
+# coding: utf-8
+
+# # Create a passive aggressive classifier to predict TP53 mutation from gene expression data in TCGA
+
+# In[1]:
+
+import os
+import urllib
+import random
+import warnings
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn import preprocessing, grid_search
+from sklearn.cross_validation import train_test_split
```

Comment:
Thank you very much for your review.  It's a great name indeed.

The algorithm is passive (no update) when the example is classified correctly and the distance is > '1 unit' from the decision boundary.

The algorithm is aggressive (updates) when the classification is wrong or the classification is correct, but the distance is < '1 unit' from the decision boundary.   Loss function is hinge or hinge^2.  The update is proportional to the loss function (instead of using a learning rate).

I guess it's similar to the SVM in that examples close to the boundary affect the decision boundary far more than examples very far from the boundary.

The learning time is very fast (can be used 'online').

This [video](https://www.youtube.com/watch?v=uxGDwyPWNkU) has a pretty good explanation.